blog/27-03-2023-feature-flags-if-statements.md

```diff
@@ -0,0 +1,64 @@
+---
+slug: 'feature-flags-if-statements'
+title: 'Feature Flags are just if Statements, Right?'
+date: 2023-03-27
+authors: [agardnerit]
+description: 'Feature Flags are Just if Statements, Right?'
+keywords: [feature flags, openfeature]
+tags: [feature flags]
+draft: false
+---
+
+Most feature flag explainers begin by explaining that feature flags are glorified if statements.
+While that’s true (to a point), feature flags wouldn’t be as versatile or as popular as they are, if that’s all they were. Indeed, you wouldn’t need a feature flag – you could easily achieve the same thing with an environment variable.
+
+<!--truncate-->
+
+<img src={require('@site/static/img/blog/feature-flags-if-statements/feature-flag-if.png').default} />
+
+In code, the above would look like this:
+
+```shell
+if isPremiumMember {
+  return v2
+} else {
+  return v1
+}
+```
+Now try modelling this in code:
+- If user is premium and located in Europe = v2 (purple)
+- If user is premium and located in the US = v3 (red)
+- If user is premium and located in Australia = v4 (orange)
+- Everyone else gets v1 (green)
+That’s 2 variables. Imagine how many variations a real system has. Now imagine trying to update these conditions in real time, without a redeployment.
+
+<img src={require('@site/static/img/blog/feature-flags-if-statements/feature-flag-context.png').default} />
+
+The real power of feature flags:
+- They can be as simple or as complex as you like (this makes getting started easy with plenty of room to grow and expand)
```

Comment:
The first point seems to be a proper sentence, it has a subject ("They can be...") while the others are sentence fragments and don't include the subject "they". This makes the points feel a bit disjointed.